include/stdexec/execution.hpp

```diff
@@ -5736,13 +5736,428 @@ namespace stdexec {
+          {
+            using _ValueType = decay_t<_T>;
+            using _Alloc = typename 
+                std::allocator_traits<_Allocator>::template rebind_alloc<_ValueType>;
+            _Alloc __alloc{__allocator_};
+            _ValueType* __pointer = std::allocator_traits<_Alloc>::allocate(__alloc, 1);
+            try {
+              std::allocator_traits<_Alloc>::construct(__alloc, __pointer, (_T&&) __object);
+            } catch (...) {
+              std::allocator_traits<_Alloc>::deallocate(__alloc, __pointer, 1);
+              throw;
+            }
+            __object_pointer_ = __pointer;
+          }
+
+        template <class _T, class... Args>
+        __t(std::in_place_type_t<_T>, Args&&... __args)
+        : __vtable_{&__storage_vtbl<__t, decay_t<_T>, _Vtable, __delete_t(void(*)() noexcept)>}
+        {
+            using _ValueType = decay_t<_T>;
+            using _Alloc = typename 
+                std::allocator_traits<_Allocator>::template rebind_alloc<_ValueType>;
+            _Alloc __alloc{__allocator_};
+            _ValueType* __pointer = std::allocator_traits<_Alloc>::allocate(__alloc, 1);
+            try {
+              std::allocator_traits<_Alloc>::construct(__alloc, __pointer, (Args&&) __args...);
+            } catch (...) {
+              std::allocator_traits<_Alloc>::deallocate(__alloc, __pointer, 1);
+              throw;
+            }
+            __object_pointer_ = __pointer;
+        }
+
+        __t(const __t&) = delete;
+        __t& operator=(const __t&) = delete;
+
+        __t(__t&& other) noexcept
+        : __vtable_{std::exchange(other.__vtable_, nullptr)}
+        , __object_pointer_{std::exchange(other.__vtable_, nullptr)} {}
+
+        __t& operator=(__t&& other) noexcept {
+          __vtable_= std::exchange(other.__vtable_, nullptr);
+          __object_pointer_ = std::exchange(other.__vtable_, nullptr);
+          return *this;
+        }
+
+        ~__t() {
+          __reset();
+        }
+
+        void __reset() noexcept {
+          if (__vtable_) {
+            (*__vtable_)(__delete, this);
+            __object_pointer_ = nullptr;
+            __vtable_ = nullptr;
+          }
+        }
+
+       private:
+        friend const _Vtable* tag_invoke(__get_vtable_t, const __t& __self) noexcept {
+          return __self.__vtable_;
+        }
+
+        friend void* tag_invoke(__get_object_pointer_t, const __t& __self) noexcept {
+          return __self.__object_pointer_;
+        }
+
+        template <class _T>
+          friend void tag_invoke(__delete_t, __type<_T>, __t& __self) noexcept {
+            using _Alloc = typename 
+                std::allocator_traits<_Allocator>::template rebind_alloc<_T>;
+            if (__self.__object_pointer_) {
+              _Alloc __alloc{__self.__allocator_};
+              _T* __pointer = reinterpret_cast<_T*>(std::exchange(__self.__object_pointer_, nullptr));
```

Comment:
I'm pretty sure this can be a `static_cast`.

Reply:
Correct